OrbitCore/CallstackData.cpp

```diff
@@ -20,8 +25,88 @@ void CallstackData::AddUniqueCallStack(CallStack call_stack) {
   unique_callstacks_[hash] = std::make_shared<CallStack>(std::move(call_stack));
 }
 
-void CallstackData::AddCallStackFromKnownCallstackData(
-    const orbit_client_protos::CallstackEvent& event, const CallstackData* known_callstack_data) {
+uint32_t CallstackData::GetCallstackEventsCount() const {
+  uint32_t count = 0;
+  for (const auto& tid_and_events : callstack_events_by_tid_) {
+    count += tid_and_events.second.size();
+  }
+  return count;
+}
+
+absl::flat_hash_map<int32_t, uint32_t> CallstackData::GetCallstackEventsCountsPerTid() const {
```

Comment:
That is creating a copy, which should reflect in the name. Where is this used?
Also you should add the mutex, right?

Reply:
I don't quite get why it does not make a copy? it is essentially returning a new map with new elements, right? However, feel free to keep the name, I don't have a very good idea though.

Reply:
I just meant that the new object is not the same as an existing one, so it's not a copy, it's just a new object.

Reply:
okay. agree.